src/NHibernate/Bytecode/Lightweight/ReflectionOptimizer.cs

```diff
@@ -45,7 +56,26 @@ public ReflectionOptimizer(System.Type mappedType, IGetter[] getters, ISetter[]
 			GetPropertyValuesInvoker getInvoker = GenerateGetPropertyValuesMethod(getters);
 			SetPropertyValuesInvoker setInvoker = GenerateSetPropertyValuesMethod(setters);
 
-			accessOptimizer = new AccessOptimizer(getInvoker, setInvoker, getters, setters);
+			var getMethods = new GetPropertyValueInvoker[getters.Length];
+			for (var i = 0; i < getters.Length; i++)
+			{
+				getMethods[i] = GenerateGetPropertyValueMethod(getters[i]) ?? getters[i].Get;
+			}
+
+			var setMethods = new SetPropertyValueInvoker[setters.Length];
+			for (var i = 0; i < setters.Length; i++)
+			{
+				setMethods[i] = GenerateSetPropertyValueMethod(setters[i]) ?? setters[i].Set;
+			}
+
+			accessOptimizer = new AccessOptimizer(
+				getInvoker,
+				setInvoker,
+				getMethods,
+				setMethods,
+				GenerateGetPropertyValueMethod(specializedGetter),
+				GenerateSetPropertyValueMethod(specializedSetter)
```

Comment:
I am thinking about it a bit late, but it does not look to me guaranteed that the specializedGetter/Setter will be optimizable. In such case, we need a fallback here, something like `?? specializedGetter?.Get`. Same for the setter.

Reply:
You are right, I somehow overlooked it. `specializedGetter?.Get` does not work so I added a ternary if instead with a todo for removing it once the obsolete constructor will be removed